modules/packages/LinearAlgebra.chpl

```diff
@@ -1450,37 +1450,55 @@ proc eig(A: [] ?t, param left = false, param right = false)
 
   if !left && !right {
     var vl, vr: [1..1, 1..n] t;
+    if t == complex{
+    LAPACK.geev(lapack_memory_order.row_major, 'N', 'N', copy, w, vl, vr);
```

Comment:
Where is `w` defined?

Reply:
You could re-use the existing `wr` or `wi` for `w`, but we'd be wasting memory in the complex case due to creating an extra unused temp array.

We might be able to do something like this instead:

```chpl
var wr: [1..n] t;
var wi: if t == complex then nothing else [1..n] t; // unused for real case
```

Then our `geev` call uses `wr`:

```
if t == complex {
  LAPACK.geev(lapack_memory_order.row_major, 'N', 'N', copy, wr, vl, vr);
  eigVals = w;
}
```